--- whisperfish-store/src/schema/protocol.rs	2024-07-14 10:54:22.184549480 +0200
+++ whisperfish-store/src/schema/protocol.rs	2024-07-14 10:57:21.805193071 +0200
@@ -1,55 +1,72 @@
 // @generated automatically by Diesel CLI.
 
 diesel::table! {
+    use diesel::sql_types::*;
+    use crate::store::orm::IdentityMapping;
+
     identity_records (address, identity) {
         address -> Text,
         record -> Binary,
-        identity -> Text,
+        identity -> IdentityMapping,
     }
 }
 
 diesel::table! {
+    use diesel::sql_types::*;
+    use crate::store::orm::IdentityMapping;
+
     kyber_prekeys (id) {
         id -> Integer,
         record -> Binary,
-        identity -> Text,
+        identity -> IdentityMapping,
         is_last_resort -> Bool,
     }
 }
 
 diesel::table! {
+    use diesel::sql_types::*;
+    use crate::store::orm::IdentityMapping;
+
     prekeys (id) {
         id -> Integer,
         record -> Binary,
-        identity -> Text,
+        identity -> IdentityMapping,
     }
 }
 
 diesel::table! {
+    use diesel::sql_types::*;
+    use crate::store::orm::IdentityMapping;
+
     sender_key_records (address, device, distribution_id) {
         address -> Text,
         device -> Integer,
         distribution_id -> Text,
         record -> Binary,
         created_at -> Timestamp,
-        identity -> Text,
+        identity -> IdentityMapping,
     }
 }
 
 diesel::table! {
+    use diesel::sql_types::*;
+    use crate::store::orm::IdentityMapping;
     session_records (address, device_id, identity) {
         address -> Text,
         device_id -> Integer,
         record -> Binary,
-        identity -> Text,
+        identity -> IdentityMapping,
     }
 }
 
 diesel::table! {
+    use diesel::sql_types::*;
+    use crate::store::orm::IdentityMapping;
+
     signed_prekeys (id) {
         id -> Integer,
         record -> Binary,
-        identity -> Text,
+        identity -> IdentityMapping,
     }
 }
 
