--- whisperfish-store/src/schema.rs	2024-03-21 19:54:08.031088264 +0100
+++ whisperfish-store/src/schema.rs	2024-03-21 19:57:00.529175793 +0100
@@ -1,4 +1,7 @@
+use crate::orm::Identity;
+
 // @generated automatically by Diesel CLI.
+pub mod migrations;
 
 diesel::table! {
     attachments (id) {
@@ -99,23 +102,32 @@
 }
 
 diesel::table! {
+    use diesel::sql_types::*;
+    use crate::store::orm::IdentityMapping;
+
     identity_records (address) {
         address -> Text,
         record -> Binary,
-        identity -> Text,
+        identity -> IdentityMapping,
     }
 }
 
 diesel::table! {
+    use diesel::sql_types::*;
+    use crate::store::orm::IdentityMapping;
+
     kyber_prekeys (id) {
         id -> Integer,
         record -> Binary,
-        identity -> Text,
+        identity -> IdentityMapping,
         is_last_resort -> Bool,
     }
 }
 
 diesel::table! {
+    use diesel::sql_types::*;
+    use crate::store::orm::MessageTypeMapping;
+
     messages (id) {
         id -> Integer,
         session_id -> Integer,
@@ -141,15 +153,18 @@
         latest_revision_id -> Nullable<Integer>,
         original_message_id -> Nullable<Integer>,
         revision_number -> Integer,
-        message_type -> Nullable<Text>,
+        message_type -> Nullable<MessageTypeMapping>,
     }
 }
 
 diesel::table! {
+    use diesel::sql_types::*;
+    use crate::store::orm::IdentityMapping;
+
     prekeys (id) {
         id -> Integer,
         record -> Binary,
-        identity -> Text,
+        identity -> IdentityMapping,
     }
 }
 
@@ -205,22 +220,28 @@
 }
 
 diesel::table! {
+    use diesel::sql_types::*;
+    use crate::store::orm::IdentityMapping;
+
     sender_key_records (address, device, distribution_id) {
         address -> Text,
         device -> Integer,
         distribution_id -> Text,
         record -> Binary,
         created_at -> Timestamp,
-        identity -> Text,
+        identity -> IdentityMapping,
     }
 }
 
 diesel::table! {
+    use diesel::sql_types::*;
+    use crate::store::orm::IdentityMapping;
+
     session_records (address, device_id) {
         address -> Text,
         device_id -> Integer,
         record -> Binary,
-        identity -> Text,
+        identity -> IdentityMapping,
     }
 }
 
@@ -240,10 +261,13 @@
 }
 
 diesel::table! {
+    use diesel::sql_types::*;
+    use crate::store::orm::IdentityMapping;
+
     signed_prekeys (id) {
         id -> Integer,
         record -> Binary,
-        identity -> Text,
+        identity -> IdentityMapping,
     }
 }
 
