--- whisperfish-store/src/schema.rs	2024-01-04 16:22:47.014931237 +0100
+++ whisperfish-store/src/schema.rs	2024-01-04 16:17:19.478576228 +0100
@@ -1,4 +1,5 @@
 // @generated automatically by Diesel CLI.
+pub mod migrations;
 
 diesel::table! {
     attachments (id) {
@@ -86,10 +87,13 @@
 }
 
 diesel::table! {
+    use diesel::sql_types::*;
+    use crate::store::orm::IdentityMapping;
+
     kyber_prekeys (id) {
         id -> Integer,
         record -> Binary,
-        identity -> Text,
+        identity -> IdentityMapping,
     }
 }
 
@@ -117,10 +121,13 @@
 }
 
 diesel::table! {
+    use diesel::sql_types::*;
+    use crate::store::orm::IdentityMapping;
+
     prekeys (id) {
         id -> Integer,
         record -> Binary,
-        identity -> Text,
+        identity -> IdentityMapping,
     }
 }
 
@@ -209,10 +216,13 @@
 }
 
 diesel::table! {
+    use diesel::sql_types::*;
+    use crate::store::orm::IdentityMapping;
+
     signed_prekeys (id) {
         id -> Integer,
         record -> Binary,
-        identity -> Text,
+        identity -> IdentityMapping,
     }
 }
 
