--- whisperfish-store/src/schema.rs	2024-09-18 11:02:36.401569009 +0200
+++ whisperfish-store/src/schema.rs	2024-09-18 11:38:20.127421277 +0200
@@ -1,4 +1,7 @@
 // @generated automatically by Diesel CLI.
+pub mod migrations;
+pub mod protocol;
+pub use protocol::*;
 
 diesel::table! {
     attachments (id) {
@@ -100,6 +103,9 @@
 }
 
 diesel::table! {
+    use diesel::sql_types::*;
+    use crate::store::orm::MessageTypeMapping;
+
     messages (id) {
         id -> Integer,
         session_id -> Integer,
@@ -125,7 +131,7 @@
         latest_revision_id -> Nullable<Integer>,
         original_message_id -> Nullable<Integer>,
         revision_number -> Integer,
-        message_type -> Nullable<Text>,
+        message_type -> Nullable<MessageTypeMapping>,
         expire_timer_version -> Integer,
     }
 }
