--- a/whisperfish-store/src/schema.rs
+++ b/whisperfish-store/src/schema.rs
@@ -1,4 +1,7 @@
 // @generated automatically by Diesel CLI.
+pub mod migrations;
+pub mod protocol;
+pub use protocol::*;
 
 diesel::table! {
     attachments (id) {
@@ -41,15 +44,18 @@
 }
 
 diesel::table! {
+    use diesel::sql_types::*;
+    use crate::store::orm::{CallTypeMapping, EventTypeMapping};
+
     calls (id) {
         id -> Integer,
         call_id -> Integer,
         message_id -> Nullable<Integer>,
         session_id -> Integer,
         #[sql_name = "type"]
-        type_ -> Text,
+        type_ -> CallTypeMapping,
         is_outbound -> Bool,
-        event -> Text,
+        event -> EventTypeMapping,
         timestamp -> Timestamp,
         ringer -> Integer,
         deletion_timestamp -> Nullable<Timestamp>,
@@ -153,6 +159,9 @@
 }
 
 diesel::table! {
+    use diesel::sql_types::*;
+    use crate::store::orm::MessageTypeMapping;
+
     messages (id) {
         id -> Integer,
         session_id -> Integer,
@@ -178,7 +187,7 @@
         latest_revision_id -> Nullable<Integer>,
         original_message_id -> Nullable<Integer>,
         revision_number -> Integer,
-        message_type -> Nullable<Text>,
+        message_type -> Nullable<MessageTypeMapping>,
         expire_timer_version -> Integer,
     }
 }
